dashboard/app/controllers/levels_controller.rb

```diff
@@ -485,6 +493,58 @@ def embed_level
     render 'levels/show'
   end
 
+  # GET /levels/:id/extra_links
+  # Get the extra links for the level, for use by levelbuilders.
+  # This is used by lab2 levels that cannot use the haml "extra links" box
+  # as that box will not refresh when changing levels.
+  def extra_links
```

Comment:
I don't know if this is where this logic should all live long term. It might make more sense to return a slimmed down version that the frontend can parse. In order to get something working for levelbuilders I think it's a decent start (I was able to copy a lot of this from the haml file), but I am tracking improving it [here](https://codedotorg.atlassian.net/browse/CT-550)